pandas/core/generic.py

```diff
@@ -5783,7 +5784,7 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                   **kwargs)) - 1)
         if freq is None:
             mask = isnull(_values_from_object(self))
-            np.putmask(rs.values, mask, np.nan)
```

Comment:
can you confirm the documentation on mask is correct ? if so, I do not see the issue with fixing the previously buggy code (you confirm this `np.putmask(rs.values, mask, np.nan)` is buggy?) as otherwise `test_pct_change` fails.
Can you also confirm that your proposal `rs[mask] = np.nan` is not the equivalent to `rs.mask(mask, np.nan,inplace!True)` which is equivalent to the original `np.putmask(rs.values, mask, np.nan)` but without assuming rs.values is a view ?